runtime/fundamentals/testing.md

```diff
@@ -0,0 +1,739 @@
+---
+title: "Testing"
+oldUrl: 
+  - /runtime/manual/advanced/language_server/testing_api/
+  - /runtime/manual/basics/testing/
+  - /runtime/manual/basics/testing/coverage/
+  - /runtime/manual/basics/testing/assertions/
+  - /runtime/manual/basics/testing/mocking/
+  - /runtime/manual/basics/testing/behavior_driven_development
+  - /runtime/manual/testing/documentation/
+  - /runtime/manual/basics/testing/sanitizers/
+  - /runtime/manual/basics/testing/snapshot_testing/
+---
+
+Deno provides a built-in test runner for writing and running tests in both
+JavaScript and TypeScript. This makes it easy to ensure your code is reliable
+and functions as expected without needing to install any additional dependencies
+or tools.
+
+The Deno test runner fully supports typescript and async tests and allows for
+fine-grained control over permissions for each test, enhancing your code
+security.
+
+## Writing Tests
+
+To define a test in Deno, you use the `Deno.test()` function. Here are some
+examples:
+
+```ts title="my_test.ts"
+import { assertEquals } from "@std/assert";
+
+// basic test
+Deno.test("simple test", () => {
+  const x = 1 + 2;
+  assertEquals(x, 3);
+});
+
+// a named test
+Deno.test("addition test", () => {
+  const sum = 2 + 3;
+  assertEquals(sum, 5);
+});
+
+// async test:
+import { delay } from "@std/async";
+
+Deno.test("async test", async () => {
+  const x = 1 + 2;
+  await delay(100);
+  assertEquals(x, 3);
+});
+
+// test with permissions
+Deno.test({
+  name: "read file test",
+  permissions: { read: true },
+  fn: () => {
+    const data = Deno.readTextFileSync("./somefile.txt");
+    assertEquals(data, "expected content");
+  },
+});
+```
```

Comment:
Nit: we should maybe hint that there is also a jest-like `expect` API in std via https://jsr.io/@std/expect for folks who prefer that. Let's add one example which uses that.

Reply:
added <3